pallets/spin-polkadot/src/lib.rs

```diff
@@ -0,0 +1,291 @@
+        type CoolDownPeriodBlocks: Get<BlockNumberFor<Self>>;
+
+        #[pallet::constant]
+        type MaxVotesAncestries: Get<u32>;
+    }
+
+    #[pallet::pallet]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Clone, PartialEq)]
+    pub enum SlowchainState<BlockNumber: Clone + PartialEq> {
+        Operational {
+            last_alive_message_block_number: BlockNumber,
+        },
+        CoolDown {
+            start_block_number: BlockNumber,
+        },
+    }
+
+    /// State of the slowchain: Operational, CoolDown or None
+    #[pallet::storage]
+    pub type State<T: Config> = StorageValue<_, SlowchainState<BlockNumberFor<T>>>;
+
+    /// Set of fastchain validators used to verify alive messages and elect a leader
+    #[pallet::storage]
+    pub type ValidatorSet<T: Config> =
+        StorageMap<_, Blake2_128Concat, T::AccountId, (), ValueQuery>;
+
+    /// Last seen alive message
+    #[pallet::storage]
+    pub type LastAliveMessage<T: Config> =
+        StorageValue<_, AliveMessageProof<T::MaxVotesAncestries, HeaderFor<T>>>;
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// A new alive message has been received.
+        HeartbeatReceived {
+            block_number: BlockNumberFor<T>,
+            who: T::AccountId,
+        },
+        /// `CoolDown` mode has been triggered.
+        StartedCoolDown {
+            block_number: BlockNumberFor<T>,
+            last_alive_message_block_number: BlockNumberFor<T>,
+        },
+        /// `CoolDown` mode has ended.
+        FinishedCoolDown { block_number: BlockNumberFor<T> },
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        IntegerOverflow,
+        CoolDownPeriod,
+        BlockNumberDecreased,
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+        fn on_initialize(current_block_number: BlockNumberFor<T>) -> Weight {
+            let mut weight = Weight::zero();
+            weight += T::DbWeight::get().reads(1);
+
+            match <State<T>>::get() {
+                Some(SlowchainState::Operational {
+                    last_alive_message_block_number,
+                }) => {
+                    let timeout_blocks = T::TimeoutBlocks::get();
+
+                    let deadline_block_number =
+                        match last_alive_message_block_number.checked_add(&timeout_blocks) {
+                            Some(deadline_block_number) => deadline_block_number,
+                            None => return weight,
+                        };
+
+                    if current_block_number > deadline_block_number {
+                        <State<T>>::put(SlowchainState::CoolDown {
+                            start_block_number: current_block_number,
+                        });
+                        weight += T::DbWeight::get().writes(1);
```

Comment:
You do `weight += T::DbWeight::get().writes(1);`  in all branches of [the match](https://github.com/QuantumFusion-network/qf-solochain/pull/33/files#diff-e56b4b7860804be18b05bccf32ff0cf289e80ebd7e28a396166d8d8c803a96cfR172).
So you can do  `weight += T::DbWeight::get().writes(1);` once [here](https://github.com/QuantumFusion-network/qf-solochain/pull/33/files#diff-e56b4b7860804be18b05bccf32ff0cf289e80ebd7e28a396166d8d8c803a96cfR220).

Reply:
But I have it inside condition as well. [Here](https://github.com/QuantumFusion-network/qf-solochain/blob/f000656917d594979521bed0c4af405cbc81048d/pallets/spin-polkadot/src/lib.rs#L202-L206) and [here](https://github.com/QuantumFusion-network/qf-solochain/blob/f000656917d594979521bed0c4af405cbc81048d/pallets/spin-polkadot/src/lib.rs#L184-L188)